ladders/Controllers/LaddersController.cs

```diff
@@ -0,0 +1,274 @@
+        {
+            if (id == null) return NotFound();
+
+            var ladderModel = await _context.LadderModel
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (ladderModel == null) return NotFound();
+
+            return View(ladderModel);
+        }
+
+        [ActionName("Join")]
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Join_Post(int? id)
+        {
+            if (id == null) return NotFound();
+
+            var ladderModel = await _context.LadderModel
+                .Include(ladder => ladder.ApprovalUsersList)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (ladderModel == null) return NotFound();
+
+            var me = await GetMe();
+            if (me == null) return RedirectToPage("Profile", "Create");
+
+            if (IsMember(me, ladderModel))
+                return RedirectToAction(nameof(Details), new {id});
+
+            if (ladderModel.ApprovalUsersList == null)
+                ladderModel.ApprovalUsersList = new List<ProfileModel>();
+
+            ladderModel.ApprovalUsersList.Add(me);
+
+            _context.Update(ladderModel);
+            await _context.SaveChangesAsync();
+            return RedirectToAction(nameof(Details), new {id});
+        }
+
+        // GET: Ladders/Approval/5
+        public async Task<IActionResult> Approval(int? id)
+        {
+            if (!AmIAdmin()) return RedirectToAction(nameof(Index));
+
+            if (id == null) return NotFound();
+
+            var ladderModel = await _context.LadderModel
+                .Include(m => m.ApprovalUsersList)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (ladderModel == null) return NotFound();
+
+            return View(ladderModel);
+        }
+
+        [ActionName("Approval")]
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        // GET: Ladders/Approval/5
+        public async Task<IActionResult> Approval(int? id, string userId, bool add)
+        {
+            if (!AmIAdmin()) return RedirectToAction(nameof(Index));
+
+            if (id == null || userId == null) return NotFound();
+
+            var ladderModel = await _context.LadderModel
+                .Include(m => m.ApprovalUsersList)
+                .Include(l => l.MemberList)
+                .FirstOrDefaultAsync(m => m.Id == id);
+            var user = await _context.ProfileModel.FirstOrDefaultAsync(u => u.UserId == userId);
+
+            if (ladderModel == null || user == null) return NotFound();
+
+            if (!ladderModel.ApprovalUsersList.Contains(user))
+                return View(ladderModel);
+
+            ladderModel.ApprovalUsersList.Remove(user);
+
+            if (add)
+                ladderModel.MemberList.Add(user);
+
+            _context.Update(ladderModel);
+            await _context.SaveChangesAsync();
+
+            return View(ladderModel);
+        }
+
+
+        // GET: Ladders/Create
```

Comment:
I'm pretty sure GET should not have side effects thus this should be a PUT or POST

Reply:
The GET is to show the form to create the ladder?

Reply:
I misread the new as creating a new ladder